### PR TITLE
Chroma db plugin for lm studio

### PR DESCRIPTION
**Pull request by BBC-Esq**
Am I doing this correctly?  Man, git is complicated and I work in the legal profession too...lol.

### CODE REVIEW

File: ChromaDB-Plugin-for-LM-Studio/config.yaml [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+  .docx: Docx2txtLoader
+  .eml: UnstructuredEmailLoader
+  .enex: EverNoteLoader
+  .json: JSONLoader
+  .msg: UnstructuredEmailLoader
+  .pdf: PDFMinerLoader
+  .txt: TextLoader
+  .xls: UnstructuredExcelLoader
+  .xlsx: UnstructuredExcelLoader
+EMBEDDING_MODEL_NAME: C:/PATH/Scripts/LM Search Vector Database/LM Search Vector Database_v1_5
```
Comment (yagil) [inline ChromaDB-Plugin-for-LM-Studio/config.yaml:43]: Not sure how this .yaml file is used, but this path looks like it might be specific to your machine

**Comment by yagil** [comment]
Really cool work @BBC-Esq!  Code looks good with 1 comment about machine specific path.

**Comment by BBC-Esq** [comment]
> Really cool work @BBC-Esq! Code looks good with 1 comment about machine specific path.

It looks that way because I used the program...but when you use it, it will update the config.yaml so...I was just the last person who used the program.  lol

**Comment by yagil** [comment]
> > Really cool work @BBC-Esq! Code looks good with 1 comment about machine specific path.
> 
> It looks that way because I used the program...but when you use it, it will update the config.yaml so...I was just the last person who used the program. lol When you click the choose model directory and select a directory, this is exactly when it'll update.

I see. Say I didn't have a config.yaml, would it create a new one for me when I run the program?  If config.yaml is intended to contain info that's particular to the user it's probably appropriate to avoid committing it to git

**Comment by BBC-Esq** [comment]
> > > Really cool work @BBC-Esq! Code looks good with 1 comment about machine specific path.
> > 
> > 
> > It looks that way because I used the program...but when you use it, it will update the config.yaml so...I was just the last person who used the program. lol When you click the choose model directory and select a directory, this is exactly when it'll update.
> 
> I see. Say I didn't have a config.yaml, would it create a new one for me when I run the program? If config.yaml is intended to contain info that's particular to the user it's probably appropriate to avoid committing it to git

The config.yaml has default values like the list of models that are populated into a script that don't change, as well as user selected options like where the model directory is...so it's a mixture if that makes sense.  As more options are added such as different chunk sizes, overlap, etc., the config.yaml will need to be updated as well if that makes sense.  Between variables, global variables, transferring variables between scripts (and use multiprocessing and multithreading), this is what I found to be the best way...

**Comment by yagil** [comment]
> > > > Really cool work @BBC-Esq! Code looks good with 1 comment about machine specific path.
> > > 
> > > 
> > > It looks that way because I used the program...but when you use it, it will update the config.yaml so...I was just the last person who used the program. lol When you click the choose model directory and select a directory, this is exactly when it'll update.
> > 
> > 
> > I see. Say I didn't have a config.yaml, would it create a new one for me when I run the program? If config.yaml is intended to contain info that's particular to the user it's probably appropriate to avoid committing it to git
> 
> The config.yaml has default values like the list of models that are populated into a script that don't change, as well as user selected options like where the model directory is...so it's a mixture if that makes sense. As more options are added such as different chunk sizes, overlap, etc., the config.yaml will need to be updated as well if that makes sense. Between variables, global variables, transferring variables between scripts (and use multiprocessing and multithreading), this is what I found to be the best way...

Got it, sounds fine

**Review by yagil — APPROVED** [review]

**Comment by BBC-Esq** [comment]
> > > > > Really cool work @BBC-Esq! Code looks good with 1 comment about machine specific path.
> > > > 
> > > > 
> > > > It looks that way because I used the program...but when you use it, it will update the config.yaml so...I was just the last person who used the program. lol When you click the choose model directory and select a directory, this is exactly when it'll update.
> > > 
> > > 
> > > I see. Say I didn't have a config.yaml, would it create a new one for me when I run the program? If config.yaml is intended to contain info that's particular to the user it's probably appropriate to avoid committing it to git
> > 
> > 
> > The config.yaml has default values like the list of models that are populated into a script that don't change, as well as user selected options like where the model directory is...so it's a mixture if that makes sense. As more options are added such as different chunk sizes, overlap, etc., the config.yaml will need to be updated as well if that makes sense. Between variables, global variables, transferring variables between scripts (and use multiprocessing and multithreading), this is what I found to be the best way...
> 
> Got it, sounds fine

Yeah, don't know how it is in typescript, but my amateurish understanding of Python is that multithreading/multiprocessing and sharing of variables is...shall we say...complicated.  Originally I didn't even use a config file and other scripts like LocalGPT or PrivateGPT use global variables...but it's a lot more difficult to expand the code without a config file...again, this is not my main profession so take with a grain of salt.  lol

**Pull request merged**